_posts/2020-04-15-#18521.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: pr
+date: 2020-04-15
+title: "fuzz: Add process_messages harness"
+pr: 18521
+authors: [MarcoFalke]
+components: [tests]
+host: MarcoFalke
+status: upcoming
+commit:
+---
+
+## Notes
+
+- A few weeks ago, [we looked at](/17860.html) how fuzzing can find consensus bugs, such as money printing (brrrr). This
+  week we will use fuzzing to find a remote crasher.
+
+- [BIP 37](https://github.com/bitcoin/bips/blob/master/bip-0037.mediawiki) describes a way to filter transaction relay
+  by bloom filters. For the purposes of this pull request we will only need to know the different message types it
+  introduces and how the Bitcoin Core node processes them.
+
+- [CVE-2013-5700](https://en.bitcoin.it/wiki/Common_Vulnerabilities_and_Exposures#CVE-2013-5700) was a vulnerability
+  that could crash a node by merely sending two P2P messages.
+
+- There is [documentation on how to compile with AFL and libFuzzer on
+  Linux](https://github.com/bitcoin/bitcoin/blob/master/doc/fuzzing.md). Other fuzz engines and operating systems might
+  work, but are currently undocumented.
+
+- [CNode](https://doxygen.bitcoincore.org/class_c_node.html) is the data structure to represent a connection in Bitcoin
+  Core.
+
+- CConnman is the connection manager in Bitcoin Core. A thread to handle messages from connections is created at
```

Comment:
maybe [CConnman](https://doxygen.bitcoincore.org/class_c_connman.html) e.g. adding a link to https://doxygen.bitcoincore.org/class_c_connman.html